### PR TITLE
Add documentation for overall architecture

### PR DESCRIPTION
**Pull request by jiangliu**
Add documentation for overall architecture

Signed-off-by: Jiang Liu <gerry@linux.alibaba.com>

### CODE REVIEW

File: docs/nydus-dev.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+## Architecture Overview
```
Comment (imeoer) [inline docs/nydus-dev.md:1]: Great job! Maybe we can reference it from README.md.

Comment (jiangliu) [reply docs/nydus-dev.md:1]: Let's add more content to the nydus-dev doc, the add reference to it from README.md?

File: docs/nydus-dev.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+Now the rust async IO ecosystem is much more mature, and is suitable for IO intensive workloads like Nydus.
+It uses the most popular tokio async io framework. Tokio supports two types of runtime engines: multi-threaded and current thread.
+The tokio multi-threaded Runtime will automatically dispatch work items among worker threads. 
+And the tokio current thread Runtime runs pending work items by current thread, and doesn't dispatch work items to different worker threads.
+
+On the other hand, tokio-uring to do asynchronous IO by using io-uring only supports tokio current thread Runtime engine.
+Thus tokio current thread Runtime is adopted to support tokio-uring in future. 
+And an asynchronous Multi-Producer, Multi-Consumer channel is used to dispatch work items among different tokio current thread Runtimes.
+
+TODO: enable io-uring for asynchronous IO. It's challenge due to data buffer lifetime management.
```
Comment (imeoer) [inline docs/nydus-dev.md:58]: `It's a challenge`

**Review by imeoer — APPROVED** [review]

File: docs/nydus-dev.md (hunk 2 above)
Comment (jiangliu) [reply docs/nydus-dev.md]: fixed

**Pull request merged**